service/history/workflow/retry.go

```diff
@@ -63,33 +100,25 @@ func getBackoffInterval(
 	// (we made 1 non-retry attempt already) and condition (currentAttemptCounterValue+1 > maxAttempts) will be false.
 	// With 2nd retry, currentAttemptCounterValue will be 2 (1 non-retry + 1 retry attempt already made) and
 	// condition (currentAttemptCounterValue+1 > maxAttempts) will be true (means stop retrying, we tried 2 times already).
-	if maxAttempts > 0 && currentAttemptCounterValue+1 > maxAttempts {
+	if maxAttempts > 0 && currentAttempt+1 > maxAttempts {
 		return backoff.NoBackoff, enumspb.RETRY_STATE_MAXIMUM_ATTEMPTS_REACHED
 	}
 
-	maxIntervalSeconds := int64(timestamp.DurationValue(maxInterval).Round(time.Second).Seconds())
-	nextIntervalSeconds := int64(timestamp.DurationValue(initInterval).Seconds() * math.Pow(backoffCoefficient, float64(currentAttemptCounterValue-1)))
-	if nextIntervalSeconds <= 0 {
-		// math.Pow() could overflow
-		if maxIntervalSeconds > 0 {
-			nextIntervalSeconds = maxIntervalSeconds
-		} else {
-			return backoff.NoBackoff, enumspb.RETRY_STATE_TIMEOUT
-		}
-	}
-
-	if maxIntervalSeconds > 0 && nextIntervalSeconds > maxIntervalSeconds {
-		// cap next interval to MaxInterval
-		nextIntervalSeconds = maxIntervalSeconds
+	interval := time.Duration(int64(float64(initInterval.Nanoseconds()) * math.Pow(backoffCoefficient, float64(currentAttempt-1))))
```

Comment:
`initInterval` can be `nil` here and it will panic. Worth to add test and pass `nil` to all pointer parameters.